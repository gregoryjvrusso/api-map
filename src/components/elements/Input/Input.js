import styled from 'styled-components'
import InputMask from 'react-input-mask';
import { width } from 'styled-system'

const Input = styled(InputMask)`
  padding: 5px;
  ${width}
  margin: 8px 0px;
  box-sizing: border-box; 
`
export default Input