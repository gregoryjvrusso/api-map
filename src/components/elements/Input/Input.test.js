import React from 'react'
import renderer from 'react-test-renderer'
import Input from './Input.js'

describe('Input component', () => {
  it('deve conseguir renderizar [snapshot]', async () => {
    const render = renderer
    .create(<Input mask='99999-999' maskChar='' />)
    .toJSON()
    expect(render).toMatchSnapshot()
  })
  it('deve conseguir renderizar [snapshot] com width de 80 px', async () => {
    const render = renderer
    .create(<Input width='80px'/>)
    .toJSON()
    expect(render).toMatchSnapshot()
  })
})
