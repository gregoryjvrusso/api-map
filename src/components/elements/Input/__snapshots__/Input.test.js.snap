// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component deve conseguir renderizar [snapshot] 1`] = `
<input
  className="sc-bdVaJa fUlNye"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onPaste={[Function]}
/>
`;

exports[`Input component deve conseguir renderizar [snapshot] com width de 80 px 1`] = `
<input
  className="sc-bdVaJa dfgRgk"
  onBlur={[Function]}
  onFocus={[Function]}
  width="80px"
/>
`;
